### PR TITLE
#9 create backend basic configure form

### PR DESCRIPTION
**Pull request by stroiaeduard**
- Saving settings available
- Examples provided

### CODE REVIEW

File: wirecardpaymentgateway.php [hunk 1]
```diff
@@ -69,4 +70,533 @@ public function uninstall()
...
+    }
+
+    /**
+     * return options for configuration modes select
+     *
+     * @since 0.0.2
+     *
+     * @return array
+     */
+    private function getConfigurationModes()
```
Comment (jakubpolomsky) [inline wirecardpaymentgateway.php]: you can remove this method

File: wirecardpaymentgateway.php [hunk 2]
```diff
@@ -69,4 +70,533 @@ public function uninstall()
...
+
+    /**
+     * process form post
+     *
+     * @since 0.0.2
+     *
+     */
+    private function postProcess()
+    {
+        if (Tools::isSubmit('ajax')) {
```
Comment (jakubpolomsky) [inline wirecardpaymentgateway.php]: you don't need the ajax test afaik

File: wirecardpaymentgateway.php [hunk 3]
```diff
@@ -69,4 +70,533 @@ public function uninstall()
...
+                        'default' => '9abf05c1-c266-46ae-8eac-7f87ca97af28',
+                        'required' => true,
+                        'sanitize' => 'trim',
+                        'doc' => $this->l('Can set the MAID.'),
+                    ),
+                    array(
+                        'name' => 'secret',
+                        'label' => $this->l('Secret'),
+                        'type' => 'text',
+                        'default' => 'WD PayPal TEST',
```
Comment (jakubpolomsky) [inline wirecardpaymentgateway.php]: please use the correct secret as defined in the config.php in the paymentSDK-php

File: wirecardpaymentgateway.php (hunk 1 above)
Comment (stroiaeduard) [reply wirecardpaymentgateway.php]: it is used for the current version of the plugin.
I think that Justin should delete it then he change the form fields

Comment (jakubpolomsky) [reply wirecardpaymentgateway.php]: I don't see it used anywhere and it is obsolete. Remove it

File: wirecardpaymentgateway.php [hunk 4]
```diff
@@ -69,4 +70,533 @@ public function uninstall()
...
+        $helper->currentIndex = $this->context->link->getAdminLink('AdminModules', false)
+            . '&configure=' . $this->name . '&tab_module=' . $this->tab . '&module_name=' . $this->name;
+        $helper->token = Tools::getAdminTokenLite('AdminModules');
+
+        $helper->tpl_vars = array(
+            'fields_value' => $this->getConfigFieldsValues(),
+            'languages' => $this->context->controller->getLanguages(),
+            'id_language' => $this->context->language->id,
+            'ajax_configtest_url' => $this->context->link->getAdminLink('AdminModules') . '&configure=' . $this->name
+                . '&tab_module=' . $this->tab . '&module_name=' . $this->name
```
Comment (jakubpolomsky) [inline wirecardpaymentgateway.php]: this configtest isn't used anywhere as well, please consider removing all unused code

**Pull request merged**